docs/src/commands/generate.md

```diff
@@ -0,0 +1,32 @@
+# wasm-pack build
+
+The `wasm-pack generate` command creates a new RustWasm project for you,
+using [`cargo-generate`] under the hood.
+
+It takes 2 optional parameters, template and name:
+
+```
+wasm-pack generate --template <template> --name <name>
+```
+
+The template will default to the `rustwasm/wasm-pack-template` and the name
+will default to `hello-wasm`.
+
+## Template
+
+The `wasm-pack generate` command can be given an optional template argument, e.g.:
+
+```
+wasm-pack generate --template https://github.com/rustwasm/wasm-pack-template
+```
+
+The template can be an address to a git repo that contains a [`cargo-generate`]
+template.
+
+## Name
+
+The `wasm-pack generate` command can be given and optional name argument, e.g.:
+
+```
+wasm-pack generate --name myproject
```

Comment:
I wonder if it might be better to `wasm-pack generate myproject`? That jives with `cargo new` and I think commands like `git init` as well.

Reply:
so the reason i did a flag was that you can also pass --template.. we could have name be positional tho and still have the other flags. i think conforming to similar tools makes more sense.

Reply:
Ah yeah I meant just having `--name` be a positional argument instead (but keeping `--template`). We could perhaps take a leaf out of `git clone`'s book though and do something like:

```
# generates a new project named "foo"
$ wasm-pack generate https://github.com/alexcrichton/foo

# generates a new project named "bar"
$ wasm-pack generate https://github.com/alexcrichton/foo bar
```

I don't really feel too strongly any direction, but I do think that having the name be a positional argument rather than a flag might be nice.

Reply:
i really like this.

Reply:
i think this would be a neat feature but let's put it off til another PR as i'm worried this one is getting a lil big